drivers/builtin/include/mbedtls/config_adjust_legacy_crypto.h

```diff
@@ -117,7 +117,54 @@
  */
 
 /* PSA accelerated implementations */
-#if defined(MBEDTLS_PSA_CRYPTO_CLIENT)
+#if defined(MBEDTLS_PSA_CRYPTO_C)
```

Comment:
Note: preserving the ABI wasn't necessary in 1.0 but this is not hurtful either, and anyway the whole dispatch logic will be changed to "always dispatch to PSA" at some point, so IMO there's no point spending too much time on it (I wasn't even convinced we need this forward port), so let's keep it as it is (ie, aligned with 3.6).